tools/jxltran.cc

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+#include <string>
+#include <vector>
+
+#include "lib/include/jxl/decode.h"
+#include "lib/include/jxl/decode_cxx.h"
+#include "tools/cmdline.h"
+#include "tools/file_io.h"
+
+namespace jpegxl {
+namespace tools {
+namespace {
+
+struct Args {
+  void AddCommandLineOptions(CommandLineParser* cmdline) {
+    cmdline->AddPositionalOption("INPUT", /* required = */ true,
+                                 "The JPEG XL input file.", &file_in);
+
+    cmdline->AddPositionalOption("OUTPUT", /* required = */ true,
+                                 "The JPEG XL output file.", &file_out);
+
+    cmdline->AddHelpText("\nFile format options:", 0);
+
+    cmdline->AddOptionFlag('\0', "extract",
+                           "Extract the JPEG XL codestream"
+                           " from a file in the container file format.",
+                           &extract, &SetBooleanTrue);
+  }
+
+  const char* file_in = nullptr;
+  const char* file_out = nullptr;
+
+  bool extract = false;
+};
+
+JxlDecoderStatus extract_from_jxlc_box(
```

Comment:
It doesn't matter yet at this point since there is only one action anyway, but probably we shouldn't have one function per action and a set of mutually exclusive actions, but rather one big function that parses input and writes output, with some struct of options that describes what has to be done, i.e. what has to be copied and what has to be stripped at the file format level (and I suppose later we could also have modification operations that change stuff in the image or frame headers, for codestream level stuff like lossless cropping or changing the colorspace tag of an XYB image).

Reply:
Sure, I should be able to do that today.

Reply:
Done!